packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/markdownFeatures.ts

```diff
@@ -106,13 +106,12 @@ function handleMarkdownEvent(
                     textContentRange.startOffset + 1
                 );
 
-                // set the removal range to include the typed last character.
-                const lastIndex: number = (range.endContainer as Text).length;
-                range.setEnd(range.endContainer, lastIndex);
+                const text = textContentRange.extractContents().textContent;
+                const textNode = editor.getDocument().createTextNode(text);
 
                 // extract content and put it into a new element.
-                const elementToWrap = editor.getDocument().createElement(elementTag);
-                elementToWrap.appendChild(textContentRange.extractContents());
+                const elementToWrap = wrap(textNode, elementTag);
+                range.setEnd(textContentRange.endContainer, textContentRange.endOffset + 1);
```

Comment:
I think `textContentRange.endOffset + 1` could throw exceptions if the current endOffset is the last child index and endOffset + 1 does not exist, not sure if this is a possibility for the markdown features.

But I think we could verify this using `new Position(range.endContainer, PositionType.End)` and verify that `textContentRange + 1` is lower than the position offset

Reply:
That is interesting, I'll add this check.